gnomad_qc/v4/sample_qc/create_sample_qc_metadata_ht_genomes.py

```diff
@@ -0,0 +1,129 @@
+"""
+Script to create sample QC metadata HT for genomes.
+
+This script is used to create the sample QC metadata HT for genomes. The main updates
+from v3.1 to v4.0 are:
+Sample QC annotations for HGDP + TGP subset, this script directly merges the updated
+annotations from the subset meta HT to the full meta HT.
+"""
+
+
+import argparse
+import logging
+
+import hail as hl
+from gnomad.resources.resource_utils import TableResource
+from hail.utils import new_temp_file
+
+from gnomad_qc.v3.resources.basics import meta as v3_meta
+from gnomad_qc.v4.resources.meta import meta as v4_meta
+from gnomad_qc.v4.resources.meta import meta_tsv_path as v4_meta_tsv_path
+from gnomad_qc.v4.resources.sample_qc import hgdp_tgp_meta_updated
+
+logging.basicConfig(format="%(levelname)s (%(name)s %(lineno)s): %(message)s")
+logger = logging.getLogger("Create v4.0 genomes sample QC metadata HT.")
+logger.setLevel(logging.INFO)
+
+
+_meta_versions = {
+    "4.0": TableResource(
+        path="gs://gnomad/v4.0/metadata/exomes/gnomad.exomes.v4.0.sample_qc_metadata.ht"
+    ),
+}
+
+
+def import_updated_annotations(ht: hl.Table, subset_ht: hl.Table) -> hl.Table:
+    """
+    Import updated annotations from subset meta HT to full meta HT.
+
+    :param ht: v3.1 genomes meta HT of genomes
+    :param subset_ht: Updated meta HT of HGDP + TGP subset
+    :return: Updated v4 genomes meta HT
+    """
+    # Some HGDP/TGP samples in v3.1 meta HT have a prefix of "v3.1::" in the key 's',
+    # but the project_meta.sample_id field does not have this prefix. This is used as
+    # a temp key to join the subset meta HT.
+    release_v3 = ht.filter(ht.release).count()
+    ht1 = ht.filter(ht.subsets.hgdp | ht.subsets.tgp)
+    ht1 = ht1.naive_coalesce(5).checkpoint(new_temp_file("v4.0_genomes_meta", "ht"))
+    ht2 = ht.filter(~(ht.subsets.hgdp | ht.subsets.tgp))
```

Comment:
you don't use the full HT (`ht`) again after this, right? to avoid cluttering the variable namespace, you could make this
```
ht = ht.filter(~(ht.subsets.hgdp | ht.subsets.tgp))
```